jax/_src/numpy/lax_numpy.py

```diff
@@ -2098,6 +2099,24 @@ def array(object: Any, dtype: DTypeLike | None = None, copy: bool = True,
           order: str | None = "K", ndmin: int = 0) -> Array:
   if order is not None and order != "K":
     raise NotImplementedError("Only implemented for order='K'")
+  if not isinstance(object, Array):
+    # Check if object supports any of the data exchange protocols
+    # (except dlpack, see data-apis/array-api#301). If it does,
+    # consume the object as jax array and continue (but not return) so
+    # that other array() arguments get processed against the input
+    # object.
+    #
+    # Notice that data exchange protocols define dtype in the
+    # corresponding data structures and it may not be available as
+    # object.dtype. So, we'll resolve the protocols here before
+    # evaluating object.dtype.
+    if hasattr(object, '__jax_array__'):
+      object = object.__jax_array__()
+    elif xla_extension_version >= 237:
```

Comment:
Minor comment: for consistency here, perhaps use `elif hasattr(object, "__cuda_array_interface__")` and then check the `xla_extension_version` within the block.

Reply:
Done.

Btw, the "unconsistent" implementation of using `getattr` is a tiny bit more efficient as it avoids a couple of unnecessary attribute access operations but I agree that the overhead of using `hasattr` approach is negligible.